.github/workflows/core-test.yml

```diff
@@ -31,7 +31,7 @@ jobs:
           pip install -e .[test_core]
       - name: Test core with pytest
         run: |
-          pytest -vv -ra --durations=0 --durations-min=0.001 src/spikeinterface/core | tee report.txt; test ${PIPESTATUS[0]} -eq 0 || exit 1
+          pytest -m "core" -vv -ra --durations=0 --durations-min=0.001 | tee report.txt; test $? -eq 0 || exit 1
```

Comment:
I changed this as a test that the marker collection works on windows.

This also triggered a lot of imports when only basic dependencies are used which made me change all the imports below.

I think this is good for two reasons:
1) Everyone should be able to run the core tests with the markers even if they don't have all the dependencies.
2) This works good as a test that we are not putting imports on the wrong places.